_drafts/monitoring-battery-life.md

```diff
@@ -0,0 +1,500 @@
+
+Second, for each battery voltage reading we reported and used in calculations, we sampled the voltage many times in a short period and took the average of the samples. This helped filter out any noise from high power draws and low voltage readings that might have skewed our readings. In our case, the vibe motor and backlight display were the two that would really message up our voltage readings.
+
+### Not all batteries are equal
+
+There will be both good and bad batches of batteries purchased from a vendor. Some might also be super-hero batteries and be exceptional, and others might barely hit the threshold of the minimum Ah ratings. That's just how it is, especially when projects are counting cents on their BOM.
+
+Also, as you likely already know, batteries age over time and lose capacity with the number of cycles they go through. At Pebble, we took this into account by slowly updating the battery curve over time for different revisions of the hardware and then year over year to make sure that we tried our best to account for battery aging.
+
+## Measuring power consumption of the hardware
+
+Let's start trying to build a model for how long our hardware device will last when operating on battery power. The first step to take is to measure how much power the hardware consumes at a base level. This is when the device is operating in all three major states: minimal capacity, normal capacity, and under strenuous load.
+
+Having all three of these power consumption profiles help paint a picture of how much power this piece of hardware may consume, and therefore how quickly it might drain a battery.
+
+### Power Profiles for each component
+
+The first step is to create baselines for how much power each component will consume **over time**. Although the spec sheets from the hardware vendor are great and will generally tell you power consumption - they aren’t always accurate, and different components from different batches will have different power usage characteristics.
+
+The general steps to accomplishing this are:
+
+- Ensure that your development boards have power rails so that you can isolate many of the components and you can easily attach probes to them.
+- Get a nice multimeter that can measure µA and mA.
+- Write a special firmware that instruments a single component in various power modes, and determine the current draw for each mode. This should be the simplest possible firmware and ideally only contain driver code.
+- Rinse and repeat with every component that could have a big impact on power consumption.
+
+For example, the accelerometer was a component in a wearable device that could consume a lot of power if left in the wrong state for periods of time. We wrote a firmware that would set it to different sampling rates and record the current consumption and then used this to determine at what sampling rate we could keep the accelerometer while still achieving our desired 7-day battery life.
```

Comment:
```suggestion
For example, the accelerometer is a component in a wearable device that could consume a lot of power if left in the wrong state for periods of time. We wrote firmware that would set it to different sampling rates and record the current consumption and then used this to determine at what sampling rate we could keep the accelerometer while still achieving our desired 7-day battery life.
```

Also the last sentence here runs on a bit. Consider restructuring.

Reply:
Thanks. Will update.